src/modules/ekf2/ekf2_main.cpp

```diff
@@ -2062,6 +2068,17 @@ void Ekf2::update_gps_blend_states()
 	_gps_blended_state.lon = (int32_t)(1.0E7 * lon_deg_res);
 	_gps_blended_state.alt += (int32_t)blended_alt_offset_mm;
 
+	// Take GPS heading from the highest weighted receiver that is publishing a valid .heading value
+	uint8_t gps_best_yaw_index = 0;
+
+	for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+		if (PX4_ISFINITE(_gps_state[i].yaw) && (_blend_weights[i] > best_weight)) {
```

Comment:
`_blend_weights[i] > best_weight` will never be true as `best_weight` is already set to the maximum.